### PR TITLE
UF-6022 - new Kivra invoice endpoint

### PR DESCRIPTION
**Pull request by jklingst**
Adds a new endpoint for sending invoices using digital-mail-sender and Kivra

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>se.sundsvall.dept44</groupId>
         <artifactId>dept44-service-parent</artifactId>
-        <version>3.2</version>
+        <version>3.3</version>
```
Comment (cheezi0747) [inline pom.xml:8]: 3.4 fanns inte när du skrev det här (troligen) men 3.4 finns nu :)

File: src/main/java/se/sundsvall/messaging/integration/digitalmailsender/DigitalMailSenderClient.java [hunk 2]
```diff
@@ -16,4 +18,7 @@ interface DigitalMailSenderClient {
 
     @PostMapping("/sendDigitalMail")
     ResponseEntity<DigitalMailResponse> sendDigitalMail(DigitalMailRequest request);
+
+    @PostMapping("/sendDigitalInvoice")
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/messaging/integration/digitalmailsender/DigitalMailSenderClient.java:22]: This comment applies to DigitalMailSender, but shouldn't this path be in lower-case?

https://dev.dataportal.se/rest-api-profil/url-format-och-namngivning

File: src/test/java/se/sundsvall/messaging/api/StatusAndHistoryResourceTests.java [hunk 3]
```diff
@@ -35,10 +36,16 @@ class StatusAndHistoryResourceTests {
...
     private HistoryService mockHistoryService;
+    @Mock
+    private StatisticsService mockStatisticsService;
 
-    @InjectMocks
     private StatusAndHistoryResource statusAndHistoryResource;
 
+    @BeforeEach
+    void setUp() {
+        statusAndHistoryResource = new StatusAndHistoryResource(mockHistoryService, mockStatisticsService);
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/messaging/api/StatusAndHistoryResourceTests.java:46]: Why is this necessary and why was `@InjectMocks` not a good option (instead of this)?

File: src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        @Schema(description = "The receiving account (a valid BANKGIRO or PLUSGIRO number)", example = "12345", requiredMode = REQUIRED)
+        String accountNumber) { }
+
+    @With
+    @Builder(setterPrefix = "with")
+    @Schema(name = "DigitalInvoiceFile")
+    public record File(
+
+        @OneOf(APPLICATION_PDF_VALUE)
+        @Schema(description = "Content type", allowableValues = {APPLICATION_PDF_VALUE})
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java:100]: `requiredMode = REQUIRED` ?

File: src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        String contentType,
+
+        @NotBlank
+        @Schema(description = "Content (BASE64-encoded)")
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java:104]: `requiredMode = REQUIRED` ?

File: src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        String content,
+
+        @NotBlank
+        @Schema(description = "Filename")
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java:108]: `requiredMode = REQUIRED` ?

File: src/main/java/se/sundsvall/messaging/integration/digitalmailsender/DigitalMailSenderClient.java (hunk 2 above)
Comment (jklingst) [reply src/main/java/se/sundsvall/messaging/integration/digitalmailsender/DigitalMailSenderClient.java:22]: As stated: out of scope, but may very well be a candidate for change with everything else cleanup-related going on in DMS

File: src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java (hunk 5 above)
Comment (foobar-1337) [inline src/main/java/se/sundsvall/messaging/api/model/request/DigitalInvoiceRequest.java:104]: `@ValidBase64` ?

File: src/integration-test/resources/DigitalInvoiceIT/__files/test2_internalServerErrorFromDigitalMailSender/request.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
+  "party": {
```
Comment (foobar-1337) [inline src/integration-test/resources/DigitalInvoiceIT/__files/test2_internalServerErrorFromDigitalMailSender/request.json:2]: Minor issue:

Use same indentation-length on all json-files?
(some files use indentation-length of 2 spaces, other 4 spaces)

**Comment by foobar-1337** [comment]
There are 2 code smells (sonar suggests to use parameterized tests instead).

I'm not a fan of over-used "Parameterized tests". They can definitely be used sometimes though. 
Do whatever you feel is right, but if you decide not to parameterize your tests, maybe mark them as "not an issue" in Sonar?

**Review by foobar-1337 — APPROVED** [review]

**Comment by jklingst** [comment]
Will re-do the tests as parameterized ones

**Review by cheezi0747 — DISMISSED** [review]

File: src/main/java/se/sundsvall/messaging/integration/digitalmailsender/DigitalMailSenderClient.java (hunk 2 above)
Comment (jklingst) [reply src/main/java/se/sundsvall/messaging/integration/digitalmailsender/DigitalMailSenderClient.java:22]: Done in https://github.com/Sundsvallskommun/api-service-messaging/pull/107

**Review by mha109 — APPROVED** [review]
